app/models/article.rb

```diff
@@ -73,4 +77,10 @@ def generate_published_dates
   def downcase_content_format
     self.content_format = self.content_format.downcase
   end
+
+  def create_redirect
+    unless Redirect.exists?(source_path: short_path, target_path: path)
+      Redirect.create(source_path: short_path, target_path: path)
+    end
+  end
```

Comment:
1. I'm getting errors around seeding because of the uniqueness validation for short_path on articles. How do we add things to the seeds?

2. I'm not entirely sure on how to do validations between two models without using a validator. I did that on the first pass [here](https://github.com/greggawatt/website/blob/c636f3b20aa5608f022a1d00cf6c5fcc7cb241b5/app/validators/article_validator.rb) but @bookis suggested this was the wrong way to go about it. I don't have too much experience here so any guidance would help.